tests/utils/blockchain.py

```diff
@@ -66,6 +67,24 @@ def accounts(self) -> List[AccountAPI]:
         self.__ape_accounts = test_accounts
         return test_accounts
 
+    @singledispatchmethod
+    def __getitem__(self, account_id):
+        raise NotImplementedError(f"Cannot use {type(account_id)} as account ID.")
+
+    @__getitem__.register
```

Comment:
hmm, what's this?

Reply:
It's related to `singledispatchmethod` in `functools` - https://docs.python.org/3/library/functools.html#functools.singledispatchmethod - where you can define a generic function with different implementations based on parameter types.

This is something `ape` uses for their `TestAccountManager` that I had to mimic to override. In this case, you can get an `AccountAPI` object using either an int index (e.g. `accounts[0]`) or a string checksum address (e.g. `accounts["0x..."]`).